spec/unit/outputs/elasticsearch/http_client_spec.rb

```diff
@@ -287,4 +289,70 @@
       end
     end
   end
+
+  class StoppableServer
+
+    attr_reader :port
+
+    def initialize()
+      @port = rand(65535-1024) + 1025
+      queue = Queue.new
+      @first_req_waiter = java.util.concurrent.CountDownLatch.new(1)
+      @first_request = nil
+
+      @t = java.lang.Thread.new(
+        proc do
+          begin
+            @server = WEBrick::HTTPServer.new :Port => @port, :DocumentRoot => ".",
+                     :Logger => Cabin::Channel.get, # silence WEBrick logging
+                     :StartCallback => Proc.new {
+                           queue.push("started")
+                         }
```

Comment:
It may, however, be better to let WEBrick bind to a random _available_ port instead of guessing one at random.

If we specify `:Port => 0`, then the resulting `WEBrick::HTTPServer` will bind to an _available_ port and its `config` will include and a `:Port` attribute containing the actual port that was bound.

(source: https://github.com/ruby/webrick/blob/v1.7.0/lib/webrick/server.rb#L112-L114)

```suggestion
            @server = WEBrick::HTTPServer.new :Port => 0, :DocumentRoot => ".",
                     :Logger => Cabin::Channel.get, # silence WEBrick logging
                     :StartCallback => Proc.new {
                           queue.push("started")
                         }
            @port = @server.config[:Port]
```